packages/react/src/components/Animation/Animation.tsx

```diff
@@ -96,30 +158,184 @@ class Animation extends UIComponent<WithAsProp<AnimationProps>, any> {
     name: PropTypes.string,
     delay: PropTypes.string,
     direction: PropTypes.string,
-    duration: PropTypes.string,
+    duration: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.shape({
+        hide: PropTypes.string,
+        show: PropTypes.string,
+      }),
+    ]),
     fillMode: PropTypes.string,
     iterationCount: PropTypes.string,
     keyframeParams: PropTypes.object,
     playState: PropTypes.string,
     timingFunction: PropTypes.string,
+    visible: PropTypes.bool,
+    mountOnShow: PropTypes.bool,
+    transitionOnMount: PropTypes.bool,
+    unmountOnHide: PropTypes.bool,
+    onComplete: PropTypes.func,
+    onHide: PropTypes.func,
+    onShow: PropTypes.func,
+    onStart: PropTypes.func,
+  }
+
+  static defaultProps = {
+    duration: 500,
+    visible: true,
+    mountOnShow: true,
+    transitionOnMount: false,
+    unmountOnHide: false,
+  }
+
+  nextStatus
+  timeoutId
+
+  constructor(props, state) {
+    super(props, state)
+
+    const { initial: status, next } = this.computeInitialStatuses()
+    this.nextStatus = next
+    this.state = { status }
+  }
+
+  computeInitialStatuses = () => {
+    const { visible, mountOnShow, transitionOnMount, unmountOnHide } = this.props
+
+    if (visible) {
+      if (transitionOnMount) {
+        return {
+          initial: Animation.EXITED,
+          next: Animation.ENTERING,
+        }
+      }
+      return { initial: Animation.ENTERED }
+    }
+
+    if (mountOnShow || unmountOnHide) return { initial: Animation.UNMOUNTED }
+    return { initial: Animation.EXITED }
+  }
+
+  componentDidMount() {
+    this.updateStatus()
+  }
+
+  UNSAFE_componentWillReceiveProps(nextProps) {
```

Comment:
Do not use that code from SUIR, it's buggy and incompatible with React's Strict and Concurrent modes 😞

Reply:
Will refactor it (I noticed one bug so far) or will try to use https://reactjs.org/docs/animation.html